lib/spawn.js

```diff
@@ -213,7 +213,7 @@ module.exports = function spawn(_command, _args, _options) {
       )
     )
 
-    setImmediate(() => task.launch())
```

Comment:
hum, I'm not too sure it's the right solution.

By doing that,
```js
const child = spawn('echo "hey"')

child.on('error', (err) => console.error(err))
```
the event listener won't be attached before the task start. So if it crashes, it won't trigger the event listener.

What if we do
```js
setImmediate(function () {
  try {
    task.launch()
  } catch (err) {
    result.emit('error', err)
  }
})
```
instead?

Reply:
I see your point and I can adjust it to this solution.

I see there is already a catch with some "cleaning" logic just a line below which might be nice to reuse it, otherwise we would need have it in the catch within setImmediate.
By wrapping the emit in setImmediate we can achieve the desired outcome 
`
catch (err) {
    fiber.cleanup()
    setImmediate(function() {
       result.emit('error', err)
    })
    return result
  }
`

One additional benefit of removing the setImmediate from the launch is that it will also result in correct process ids being returned.
Let me know what you think.

Reply:
That sounds like a good solution 👍